stellar/deploy-contract.js

```diff
@@ -101,35 +99,25 @@ async function processDeploy(options, config, chain) {
     await broadcast(operation, wallet, chain, 'Initialized contract', options);
 }
 
-async function processUpgrade(options, _, chain) {
-    const { contractName, contractId, privateKey, wasmPath, newWasmHash } = options;
+async function upgrade(options, _, chain) {
+    const { privateKey, wasmPath } = options;
     const { rpc, networkType } = chain;
     const networkPassphrase = getNetworkPassphrase(networkType);
+    const contractAddress = chain.contracts?.axelar_gateway.address;
 
-    if (!chain.contracts) {
-        chain.contracts = {};
+    if (!contractAddress) {
+        throw new Error('Stellar Gateway Address not found.');
     }
 
     const args = `--source ${privateKey} --rpc-url ${rpc} --network-passphrase "${networkPassphrase}"`;
 
-    let cmd;
+    let cmd = `${stellarCmd} contract install --wasm ${wasmPath} ${args}`;
+    const newWasmHash = execSync(cmd, { encoding: 'utf-8', stdio: 'pipe' }).trimEnd();
```

Comment:
Save wasmHash in the contract config during deploy and upgrade, so the change gets versioned

Reply:
got it, updated.